### ISSUE TITLE
TS2305: Module '"../../../../../app/node_modules/ts-transformer-keys"' has no exported member 'keys

### ISSUE THREAD

**Issue by LeonaYoung** [kimamula/ts-transformer-keys#26]
use ts-loder report error:
TS2305: Module '"../../../../../app/node_modules/ts-transformer-keys"' has no exported member 'keys

**Comment by kimamula** [comment]
Could you provide a repository with which I can reproduce the error?

**Comment by LeonaYoung** [comment]
Sorry, my repository is private. This is my package info:
"typescript": "^3.6.2",
"ts-loader": "^6.2.1",
"react": "^16.5.2",
"webpack": "^4.41.2",

**Comment by kimamula** [comment]
I have updated `typescript`, `ts-loader` and `webpack` packages to the latest version and test the transformation in `examples/webpack` dir in this repository.
It looks that the transformation is working fine.

I would like to know how you are using this transformer so that I can find the cause of the problem.
If you could create some minimal repro, it would be the great help.

**Comment by mlecoq** [comment]
I have the same issue when I try to build my project

After import :
```
import { keys } from 'ts-transformer-keys';
```

I am using keys of my interface:

```
   const keysOfProps = keys<ProjectCommonProperties>();

            return Object.entries<string>(project.properties)
                .filter(([key]) => !keysOfProps.map(k => k.toString()).includes(key))
                .map(([key, value]) => ({ key, value }));
```


In my Webpack config, I have used the transformer:



 ```
module: {
        rules: [
            {
                test: /\.tsx?$/,
                loader: 'ts-loader?configFile=tsconfig.webpack.json',
                exclude: [/node_modules/],
                options: {
                    getCustomTransformers: program => ({
                        before: [keysTransformer(program)]
                    })
                }
            },
            {
                test: /\.js?$/,
                loader: 'source-map-loader',
                exclude: [/node_modules/, /locales/]
            }
        ]
    },
```


My tsconfig file:
```
{
    "compilerOptions": {
        "sourceMap": true,
        "target": "es2018",
        "lib": ["es2018"],
        "outDir": "dist",
        "allowJs": true,
        "moduleResolution": "node",
        "esModuleInterop": true,
        "allowSyntheticDefaultImports": true,
        "module": "commonjs",
        "resolveJsonModule": true,
        "types": ["jest", "node"],
        "baseUrl": "."
    },
    "exclude": ["node_modules", "dist", "**/*.test.ts"]
}

```

**Comment by mlecoq** [comment]
@kimamula I have made a small example reproducing the issue:
https://github.com/mlecoq/example-ts-remark-keys-issue

you just have to run `yarn build`

**Comment by mlecoq** [comment]
the issue is caused by ` "allowJs": true,` parameter

**Comment by LeonaYoung** [comment]
https://github.com/LeonaYoung/fuguizhu
This repository can reproduce the problem. 
run npm start  or npm run build

**Comment by kimamula** [comment]
Thanks for your works!
As @mlecoq mentioned, the transformer has been not working correctly when `allowJs` is `true`.
I have released v0.3.6-rc.1 which should fix the problem.
v0.3.6 will be released when https://github.com/ezolenko/rollup-plugin-typescript2/issues/191 get resolved.

**Comment by kimamula** [comment]
Released v0.4.0.
Please git it a try.
